presto-tests/src/main/java/io/prestosql/tests/AbstractTestJoinQueries.java

```diff
@@ -757,12 +757,69 @@ public void testFullJoinWithCoalesce()
                         "FULL OUTER JOIN (VALUES 1, 4, null) u(a) ON t.a = u.a " +
                         "GROUP BY 1",
                 "VALUES (1, 1), (2, 1), (3, 1), (4, 1), (40, 1), (100, 2)");
+
+        assertQuery(
+                "WITH " +
+                        "data(id, value) AS (SELECT nationkey, name FROM nation)," +
+                        "grouped AS (" +
+                        "    SELECT id, value" +
+                        "    FROM data" +
+                        "    WHERE value = 'x'" +
+                        "    GROUP BY id, value" +
+                        ")," +
+                        "joined AS (" +
+                        "    SELECT coalesce(grouped.id, data.id) id, grouped.value, data.value " +
+                        "    FROM grouped FULL JOIN data ON grouped.id = data.id " +
+                        ") " +
+                        "SELECT joined.* " +
+                        "FROM joined JOIN (VALUES 0) t(id) ON joined.id = t.id",
+                "VALUES (0, NULL, 'ALGERIA')");
+
+        // same as above but with the order of the tables in the join flipped
```

Comment:
it would be great if you add a comment which branch from this `if` chain `if (left.isVariable() && right.isVariable()) {` does it test